service/history/workflow/task_generator.go

```diff
@@ -102,6 +103,7 @@ type (
 		namespaceRegistry namespace.Registry
 		mutableState      MutableState
 		config            *configs.Config
+		logger            log.Logger
```

Comment:
why do it need logger here?

Reply:
Probably from an older revision. Removed, thanks for spotting!